Fix ComfyUI 0.3.69 compatibility issues

EnragedAntelope opened the pull request:
- Update imports to use comfy.ldm.flux.layers instead of comfy.ldm.chroma.layers for DoubleStreamBlock and SingleStreamBlock (they are now set to None in chroma.layers)
- Ensure inner_sample accepts latent_shapes parameter as required by new ComfyUI API

Fixes:
- TypeError: NoneType takes no arguments (import issue)
- TypeError: NAGCFGGuider.inner_sample() got an unexpected keyword argument 'latent_shapes'

claude committed:
Fix ComfyUI 0.3.69+ compatibility issues

- Update imports to use comfy.ldm.flux.layers instead of comfy.ldm.chroma.layers
  for DoubleStreamBlock and SingleStreamBlock (they are now set to None in chroma.layers)
- Pass latent_shapes parameter through executor.execute in NAGCFGGuider.sample()
  to match base CFGGuider implementation in latest ComfyUI

Fixes:
- TypeError: NoneType takes no arguments (import issue)
- TypeError: NAGCFGGuider.inner_sample() got an unexpected keyword argument 'latent_shapes'

zpengcom reviewed (commented):
Repaired successfully

EnragedAntelope commented:
@ChenDarYen I confirmed this works on my implementation and so did @zpengcom  - can you please merge this PR?

suede299 commented:
However, an error will still be reported after interrupting the queue.
AttributeError: 'NoneType' object has no attribute 'detach'

claude committed:
Fix AttributeError on queue interruption in Chroma model

Add None check for guidance parameter before calling .detach() to prevent
AttributeError: 'NoneType' object has no attribute 'detach' when queue
is interrupted.

When guidance is None, use zeros_like(distill_timestep) as fallback,
matching the pattern used in other model implementations (flux, hunyuan_video).

EnragedAntelope commented:
> However, an error will still be reported after interrupting the queue. AttributeError: 'NoneType' object has no attribute 'detach'

Updated, try again please?

suede299 commented:
> ed, try ag

10:09:29.943 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\ComfyUI\custom_nodes\ComfyUI-NAG\chroma\model.py", line 63, in forward_orig
10:09:29.943 [Warning] [ComfyUI-0/STDERR]     ids_negative = torch.cat((txt_ids_negative, img_ids[-origin_bsz:]), dim=1)
10:09:29.954 [Warning] [ComfyUI-0/STDERR] IndexError: Dimension out of range (expected to be in range of [-1, 0], but got 1)
When adding a new queue after interrupting the queue

EnragedAntelope commented:
@suede299 You can tell I am depending on Claude to do this! I just updated the PR again and Claude claims it is fixed. When able, please retry once more. I do not experience your error, so can't tell if this fixes it for you.

suede299 commented:
> @suede299 You can tell I am depending on Claude to do this! I just updated the PR again and Claude claims it is fixed. When able, please retry once more. I do not experience your error, so can't tell if this fixes it for you.

12:02:02.035 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\ComfyUI\custom_nodes\ComfyUI-NAG\chroma\model.py", line 230, in forward
12:02:02.035 [Warning] [ComfyUI-0/STDERR]     out = self.forward_orig(
12:02:02.036 [Warning] [ComfyUI-0/STDERR]         img, img_ids, context, txt_ids, txt_ids_negative, timestep, guidance, control, transformer_options,
12:02:02.036 [Warning] [ComfyUI-0/STDERR]         attn_mask=kwargs.get("attention_mask", None),
12:02:02.037 [Warning] [ComfyUI-0/STDERR]     )
12:02:02.037 [Warning] [ComfyUI-0/STDERR] TypeError: Chroma.forward_orig() got multiple values for argument 'attn_mask'
--------------------
nag cfg guider cannot be used at present. Not limited to after the queue is terminated.

claude committed:
Fix TypeError and add defensive checks for queue interruption

1. Fix TypeError: Chroma.forward_orig() got multiple values for argument 'attn_mask'
   - Remove incorrect txt_ids_negative parameter from else block
   - The else block calls base Chroma.forward_orig which doesn't accept txt_ids_negative
   - Previous fix in e751545 was based on misunderstanding of the code flow

2. Add defensive checks for IndexError during queue interruption
   - Check img_ids is not None and has correct dimensions before concatenation
   - Handle edge case where origin_bsz <= 0
   - Prevents "Dimension out of range" error when img_ids has unexpected shape
   - Follows same pattern as flux/model.py

This properly handles queue interruption scenarios while maintaining compatibility
with base Chroma class in non-NAG mode.

EnragedAntelope commented:
Tried again - if you are able to give it another go.

suede299 commented:
> Tried again - if you are able to give it another go.
It returned to the previous state. That is, an error is reported after the queue is interrupted and a new queue is added. The following is the complete print
--------------------
13:36:13.781 [Warning] [ComfyUI-0/STDERR] Traceback (most recent call last):
13:36:13.781 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\ComfyUI\execution.py", line 510, in execute
13:36:13.782 [Warning] [ComfyUI-0/STDERR]     output_data, output_ui, has_subgraph, has_pending_tasks = await get_output_data(prompt_id, unique_id, obj, input_data_all, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb, hidden_inputs=hidden_inputs)
13:36:13.782 [Warning] [ComfyUI-0/STDERR]                                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
13:36:13.782 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\ComfyUI\execution.py", line 324, in get_output_data
13:36:13.783 [Warning] [ComfyUI-0/STDERR]     return_values = await _async_map_node_over_list(prompt_id, unique_id, obj, input_data_all, obj.FUNCTION, allow_interrupt=True, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb, hidden_inputs=hidden_inputs)
13:36:13.783 [Warning] [ComfyUI-0/STDERR]                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
13:36:13.784 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\ComfyUI\execution.py", line 298, in _async_map_node_over_list
13:36:13.784 [Warning] [ComfyUI-0/STDERR]     await process_inputs(input_dict, i)
13:36:13.784 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\ComfyUI\execution.py", line 286, in process_inputs
13:36:13.785 [Warning] [ComfyUI-0/STDERR]     result = f(**inputs)
13:36:13.785 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\ComfyUI\comfy_extras\nodes_custom_sampler.py", line 835, in sample
13:36:13.786 [Warning] [ComfyUI-0/STDERR]     samples = guider.sample(noise.generate_noise(latent), latent_image, sampler, sigmas, denoise_mask=noise_mask, callback=callback, disable_pbar=disable_pbar, seed=noise.seed)
13:36:13.786 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\ComfyUI\custom_nodes\ComfyUI-NAG\samplers.py", line 193, in sample
13:36:13.787 [Warning] [ComfyUI-0/STDERR]     output = executor.execute(
13:36:13.787 [Warning] [ComfyUI-0/STDERR]         noise,
13:36:13.788 [Warning] [ComfyUI-0/STDERR]     ...<7 lines>...
13:36:13.788 [Warning] [ComfyUI-0/STDERR]         latent_shapes=latent_shapes,
13:36:13.788 [Warning] [ComfyUI-0/STDERR]     )
13:36:13.789 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\ComfyUI\comfy\patcher_extension.py", line 112, in execute
13:36:13.789 [Warning] [ComfyUI-0/STDERR]     return self.original(*args, **kwargs)
13:36:13.789 [Warning] [ComfyUI-0/STDERR]            ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
13:36:13.790 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\ComfyUI\comfy\samplers.py", line 997, in outer_sample
13:36:13.790 [Warning] [ComfyUI-0/STDERR]     output = self.inner_sample(noise, latent_image, device, sampler, sigmas, denoise_mask, callback, disable_pbar, seed, latent_shapes=latent_shapes)
13:36:13.791 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\ComfyUI\custom_nodes\ComfyUI-NAG\samplers.py", line 120, in inner_sample
13:36:13.791 [Warning] [ComfyUI-0/STDERR]     samples = executor.execute(
13:36:13.791 [Warning] [ComfyUI-0/STDERR]         self,
13:36:13.792 [Warning] [ComfyUI-0/STDERR]     ...<6 lines>...
13:36:13.792 [Warning] [ComfyUI-0/STDERR]         disable_pbar,
13:36:13.793 [Warning] [ComfyUI-0/STDERR]     )
13:36:13.793 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\ComfyUI\comfy\patcher_extension.py", line 112, in execute
13:36:13.793 [Warning] [ComfyUI-0/STDERR]     return self.original(*args, **kwargs)
13:36:13.794 [Warning] [ComfyUI-0/STDERR]            ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
13:36:13.794 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\ComfyUI\comfy\samplers.py", line 752, in sample
13:36:13.794 [Warning] [ComfyUI-0/STDERR]     samples = self.sampler_function(model_k, noise, sigmas, extra_args=extra_args, callback=k_callback, disable=disable_pbar, **self.extra_options)
13:36:13.795 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\python_embeded\Lib\site-packages\torch\utils\_contextlib.py", line 120, in decorate_context
13:36:13.795 [Warning] [ComfyUI-0/STDERR]     return func(*args, **kwargs)
13:36:13.795 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\ComfyUI\custom_nodes\ComfyUI-ClybsChromaNodes\clyb_Samplers.py", line 161, in sample_clyb_bdf
13:36:13.796 [Warning] [ComfyUI-0/STDERR]     return sampler_clyb_bdf(model, x, sigmas, extra_args=extra_args, callback=callback, disable=disable, scalar=scalar, eta=eta, s_noise=s_noise, noise_sampler=noise_sampler, flow=flow)
13:36:13.796 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\python_embeded\Lib\site-packages\torch\utils\_contextlib.py", line 120, in decorate_context
13:36:13.796 [Warning] [ComfyUI-0/STDERR]     return func(*args, **kwargs)
13:36:13.799 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\ComfyUI\custom_nodes\ComfyUI-ClybsChromaNodes\clyb_Samplers.py", line 108, in sampler_clyb_bdf
13:36:13.800 [Warning] [ComfyUI-0/STDERR]     first_denoised = prev_denoised if (prev_denoised is not None and sigma_down > 0) else model(x, sigmas[i] * s_in, **extra_args)
13:36:13.800 [Warning] [ComfyUI-0/STDERR]                                                                                           ~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
13:36:13.800 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\ComfyUI\comfy\samplers.py", line 401, in __call__
13:36:13.801 [Warning] [ComfyUI-0/STDERR]     out = self.inner_model(x, sigma, model_options=model_options, seed=seed)
13:36:13.801 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\ComfyUI\custom_nodes\ComfyUI-NAG\samplers.py", line 103, in __call__
13:36:13.802 [Warning] [ComfyUI-0/STDERR]     return self.predict_noise(*args, **kwargs)
13:36:13.802 [Warning] [ComfyUI-0/STDERR]            ~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
13:36:13.802 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\ComfyUI\comfy\samplers.py", line 963, in predict_noise
13:36:13.803 [Warning] [ComfyUI-0/STDERR]     return sampling_function(self.inner_model, x, timestep, self.conds.get("negative", None), self.conds.get("positive", None), self.cfg, model_options=model_options, seed=seed)
13:36:13.803 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\ComfyUI\comfy\samplers.py", line 381, in sampling_function
13:36:13.804 [Warning] [ComfyUI-0/STDERR]     out = calc_cond_batch(model, conds, x, timestep, model_options)
13:36:13.804 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\ComfyUI\comfy\samplers.py", line 206, in calc_cond_batch
13:36:13.804 [Warning] [ComfyUI-0/STDERR]     return _calc_cond_batch_outer(model, conds, x_in, timestep, model_options)
13:36:13.805 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\ComfyUI\comfy\samplers.py", line 214, in _calc_cond_batch_outer
13:36:13.805 [Warning] [ComfyUI-0/STDERR]     return executor.execute(model, conds, x_in, timestep, model_options)
13:36:13.805 [Warning] [ComfyUI-0/STDERR]            ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
13:36:13.806 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\ComfyUI\comfy\patcher_extension.py", line 112, in execute
13:36:13.806 [Warning] [ComfyUI-0/STDERR]     return self.original(*args, **kwargs)
13:36:13.807 [Warning] [ComfyUI-0/STDERR]            ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
13:36:13.807 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\ComfyUI\comfy\samplers.py", line 326, in _calc_cond_batch
13:36:13.808 [Warning] [ComfyUI-0/STDERR]     output = model.apply_model(input_x, timestep_, **c).chunk(batch_chunks)
13:36:13.808 [Warning] [ComfyUI-0/STDERR]              ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^
13:36:13.808 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\ComfyUI\comfy\model_base.py", line 161, in apply_model
13:36:13.809 [Warning] [ComfyUI-0/STDERR]     return comfy.patcher_extension.WrapperExecutor.new_class_executor(
13:36:13.809 [Warning] [ComfyUI-0/STDERR]            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
13:36:13.810 [Warning] [ComfyUI-0/STDERR]     ...<2 lines>...
13:36:13.810 [Warning] [ComfyUI-0/STDERR]         comfy.patcher_extension.get_all_wrappers(comfy.patcher_extension.WrappersMP.APPLY_MODEL, transformer_options)
13:36:13.811 [Warning] [ComfyUI-0/STDERR]         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
13:36:13.811 [Warning] [ComfyUI-0/STDERR]     ).execute(x, t, c_concat, c_crossattn, control, transformer_options, **kwargs)
13:36:13.811 [Warning] [ComfyUI-0/STDERR]     ~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
13:36:13.814 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\ComfyUI\comfy\patcher_extension.py", line 112, in execute
13:36:13.814 [Warning] [ComfyUI-0/STDERR]     return self.original(*args, **kwargs)
13:36:13.815 [Warning] [ComfyUI-0/STDERR]            ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
13:36:13.815 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\ComfyUI\comfy\model_base.py", line 203, in _apply_model
13:36:13.816 [Warning] [ComfyUI-0/STDERR]     model_output = self.diffusion_model(xc, t, context=context, control=control, transformer_options=transformer_options, **extra_conds)
13:36:13.816 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\python_embeded\Lib\site-packages\torch\nn\modules\module.py", line 1775, in _wrapped_call_impl
13:36:13.816 [Warning] [ComfyUI-0/STDERR]     return self._call_impl(*args, **kwargs)
13:36:13.817 [Warning] [ComfyUI-0/STDERR]            ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
13:36:13.817 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\python_embeded\Lib\site-packages\torch\nn\modules\module.py", line 1786, in _call_impl
13:36:13.817 [Warning] [ComfyUI-0/STDERR]     return forward_call(*args, **kwargs)
13:36:13.818 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\ComfyUI\custom_nodes\ComfyUI-NAG\chroma\model.py", line 238, in forward
13:36:13.818 [Warning] [ComfyUI-0/STDERR]     out = self.forward_orig(
13:36:13.818 [Warning] [ComfyUI-0/STDERR]         img, img_ids, context, txt_ids, timestep, guidance, control, transformer_options,
13:36:13.819 [Warning] [ComfyUI-0/STDERR]         attn_mask=kwargs.get("attention_mask", None),
13:36:13.819 [Warning] [ComfyUI-0/STDERR]     )
13:36:13.819 [Warning] [ComfyUI-0/STDERR]   File "H:\SwarmUI\dlbackend\comfy\ComfyUI\custom_nodes\ComfyUI-NAG\chroma\model.py", line 68, in forward_orig
13:36:13.820 [Warning] [ComfyUI-0/STDERR]     ids_negative = torch.cat((txt_ids_negative, img_ids), dim=1)
13:36:13.820 [Warning] [ComfyUI-0/STDERR] IndexError: Dimension out of range (expected to be in range of [-1, 0], but got 1)

claude committed:
Fix queue interruption handling across all NAG models

**Root cause:** When a queue with NAG active is interrupted, the restoration
of self.forward_orig and block.forward methods never executes, leaving them
in the NAG state. When a new queue starts without NAG active, the else block
calls self.forward_orig which is still the NAG version, causing parameter
misalignment and IndexError.

**Fixes applied to all models:**

1. **Chroma** (chroma/model.py):
   - Wrap NAG setup in try/finally to ensure restoration even on interruption
   - In else block, call Chroma.forward_orig directly instead of self.forward_orig
   - Add defensive checks for img_ids dimensions

2. **Flux** (flux/model.py):
   - Same try/finally protection
   - Call Flux.forward_orig directly in else block

3. **HunyuanVideo** (hunyuan_video/model.py):
   - Same try/finally protection
   - Call HunyuanVideo.forward_orig directly in else block

4. **Wan** (wan/model.py - both NAGWanModel and NAGVaceWanModel):
   - Wrap forward_orig call in try/finally
   - Initialize forward_orig_ and cross_attns_forward before if block
   - Restore in finally block only if NAG was active

This ensures that even if queue is interrupted mid-processing, all modified
methods are restored to their original state, preventing parameter misalignment
errors in subsequent queue runs.

EnragedAntelope commented:
Ok, 5th time is the charm! That whole error log was helpful. Want to give it another go? I had Claude trace each model execution and interruption path.

suede299 commented:
> Ok, 5th time is the charm! That whole error log was helpful. Want to give it another go? I had Claude trace each model execution and interruption path.

This commit worked without any issues in my tests.

EnragedAntelope commented:
Hooray! Thank you for your patience and testing. 
@ChenDarYen now for real you should be able to merge this!

NulliferBones commented:
I still get the error


!!! Exception during processing !!! 'NoneType' object has no attribute 'detach'
Traceback (most recent call last):
  File "/home/nullifer/Documents/AI/Comfy2/ComfyUI/execution.py", line 510, in execute
    output_data, output_ui, has_subgraph, has_pending_tasks = await get_output_data(prompt_id, unique_id, obj, input_data_all, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb, hidden_inputs=hidden_inputs)
                                                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nullifer/Documents/AI/Comfy2/ComfyUI/execution.py", line 324, in get_output_data
    return_values = await _async_map_node_over_list(prompt_id, unique_id, obj, input_data_all, obj.FUNCTION, allow_interrupt=True, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb, hidden_inputs=hidden_inputs)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nullifer/Documents/AI/Comfy2/ComfyUI/custom_nodes/comfyui-lora-manager/py/metadata_collector/metadata_hook.py", line 165, in async_map_node_over_list_with_metadata
    results = await original_map_node_over_list(
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nullifer/Documents/AI/Comfy2/ComfyUI/execution.py", line 298, in _async_map_node_over_list
    await process_inputs(input_dict, i)
  File "/home/nullifer/Documents/AI/Comfy2/ComfyUI/execution.py", line 286, in process_inputs
    result = f(**inputs)
             ^^^^^^^^^^^
  File "/home/nullifer/Documents/AI/Comfy2/ComfyUI/comfy_extras/nodes_custom_sampler.py", line 835, in sample
    samples = guider.sample(noise.generate_noise(latent), latent_image, sampler, sigmas, denoise_mask=noise_mask, callback=callback, disable_pbar=disable_pbar, seed=noise.seed)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nullifer/Documents/AI/Comfy2/ComfyUI/custom_nodes/ComfyUI-NAG/samplers.py", line 193, in sample
    output = executor.execute(
             ^^^^^^^^^^^^^^^^^
  File "/home/nullifer/Documents/AI/Comfy2/ComfyUI/comfy/patcher_extension.py", line 112, in execute
    return self.original(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nullifer/Documents/AI/Comfy2/ComfyUI/comfy/samplers.py", line 997, in outer_sample
    output = self.inner_sample(noise, latent_image, device, sampler, sigmas, denoise_mask, callback, disable_pbar, seed, latent_shapes=latent_shapes)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nullifer/Documents/AI/Comfy2/ComfyUI/custom_nodes/ComfyUI-NAG/samplers.py", line 120, in inner_sample
    samples = executor.execute(
              ^^^^^^^^^^^^^^^^^
  File "/home/nullifer/Documents/AI/Comfy2/ComfyUI/comfy/patcher_extension.py", line 112, in execute
    return self.original(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nullifer/Documents/AI/Comfy2/ComfyUI/comfy/samplers.py", line 752, in sample
    samples = self.sampler_function(model_k, noise, sigmas, extra_args=extra_args, callback=k_callback, disable=disable_pbar, **self.extra_options)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nullifer/Documents/AI/Comfy2/ComfyUI/venv/lib/python3.12/site-packages/torch/utils/_contextlib.py", line 116, in decorate_context
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/nullifer/Documents/AI/Comfy2/ComfyUI/comfy/k_diffusion/sampling.py", line 795, in sample_dpmpp_2m
    denoised = model(x, sigmas[i] * s_in, **extra_args)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nullifer/Documents/AI/Comfy2/ComfyUI/comfy/samplers.py", line 401, in __call__
    out = self.inner_model(x, sigma, model_options=model_options, seed=seed)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nullifer/Documents/AI/Comfy2/ComfyUI/custom_nodes/ComfyUI-NAG/samplers.py", line 103, in __call__
    return self.predict_noise(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nullifer/Documents/AI/Comfy2/ComfyUI/comfy/samplers.py", line 963, in predict_noise
    return sampling_function(self.inner_model, x, timestep, self.conds.get("negative", None), self.conds.get("positive", None), self.cfg, model_options=model_options, seed=seed)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nullifer/Documents/AI/Comfy2/ComfyUI/comfy/samplers.py", line 381, in sampling_function
    out = calc_cond_batch(model, conds, x, timestep, model_options)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nullifer/Documents/AI/Comfy2/ComfyUI/comfy/samplers.py", line 206, in calc_cond_batch
    return _calc_cond_batch_outer(model, conds, x_in, timestep, model_options)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nullifer/Documents/AI/Comfy2/ComfyUI/comfy/samplers.py", line 214, in _calc_cond_batch_outer
    return executor.execute(model, conds, x_in, timestep, model_options)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nullifer/Documents/AI/Comfy2/ComfyUI/comfy/patcher_extension.py", line 112, in execute
    return self.original(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nullifer/Documents/AI/Comfy2/ComfyUI/comfy/samplers.py", line 326, in _calc_cond_batch
    output = model.apply_model(input_x, timestep_, **c).chunk(batch_chunks)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nullifer/Documents/AI/Comfy2/ComfyUI/comfy/model_base.py", line 161, in apply_model
    return comfy.patcher_extension.WrapperExecutor.new_class_executor(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nullifer/Documents/AI/Comfy2/ComfyUI/comfy/patcher_extension.py", line 112, in execute
    return self.original(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nullifer/Documents/AI/Comfy2/ComfyUI/comfy/model_base.py", line 203, in _apply_model
    model_output = self.diffusion_model(xc, t, context=context, control=control, transformer_options=transformer_options, **extra_conds)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nullifer/Documents/AI/Comfy2/ComfyUI/venv/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1751, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nullifer/Documents/AI/Comfy2/ComfyUI/venv/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1762, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nullifer/Documents/AI/Comfy2/ComfyUI/custom_nodes/ComfyUI-NAG/chroma/model.py", line 226, in forward
    out = self.forward_orig(
          ^^^^^^^^^^^^^^^^^^
  File "/home/nullifer/Documents/AI/Comfy2/ComfyUI/custom_nodes/ComfyUI-NAG/chroma/model.py", line 41, in forward_orig
    distil_guidance = timestep_embedding(guidance.detach().clone(), 16).to(img.device, img.dtype)
                                         ^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'detach'

Prompt executed in 51.24 seconds

EnragedAntelope commented:
@NulliferBones Hi, I think you actually need to use this PR and it should fix your issue. If you only pulled the base repo, you will still see the issue until this PR is merged.

If you need a fix quickly this may work:
`git fetch origin pull/59/head:pr-59 && git checkout pr-59`

Otherwise you can wait for the merge. If you do pull the PR, then later when this is merged, you should go back to the main branch:
`git checkout main && git pull origin main`

zwukong commented:
Thanks ,can this pr also support hunyuan video 1.5

zwukong commented:
hunyuan video 1.5 error;TypeError: NAGHunyuanVideo.forward() missing 1 required positional argument: 'y'

EnragedAntelope commented:
> Thanks ,can this pr also support hunyuan video 1.5

That is a new feature request so maybe the dev will address that separately in another PR. This PR was just to get things fixed and working again on latest versions of ComfyUI.

NulliferBones commented:
> @NulliferBones Hi, I think you actually need to use this PR and it should fix your issue. If you only pulled the base repo, you will still see the issue until this PR is merged.
> 
> If you need a fix quickly this may work: `git fetch origin pull/59/head:pr-59 && git checkout pr-59`
> 
> Otherwise you can wait for the merge. If you do pull the PR, then later when this is merged, you should go back to the main branch: `git checkout main && git pull origin main`

Strange, I cloned your full updated repo and I still got the error, but this seems to have fixed it. Thanks.

EnragedAntelope commented:
Created an updated PR to address these issues and to also support ZImage Turbo (Lumina 2).
